lib/project-dependencies.js

```diff
@@ -29,7 +30,8 @@ async function collect(options, elmJson, elmVersion) {
   let hasDependenciesThatCouldNotBeDownloaded = false;
 
   const projectDepsPromises = Object.entries(dependenciesEntries).map(
-    async ([name, constraint]) => {
+    async ([pkgname, constraint]) => {
+      const name = /** @type {PackageName} */ (pkgname);
```

Comment:
(Mostly out of curiosity) Could this type potentially be inferred by typing `dependenciesEntries` instead?

Reply:
It is typed, but because of excess properties, ts can't infer it: [FAQ § Excess Properties Are OK](https://github.com/microsoft/TypeScript/wiki/FAQ#indirect-excess-properties-are-ok).